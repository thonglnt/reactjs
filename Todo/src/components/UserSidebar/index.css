.sidebar-title {
  color: var(--white);
}
