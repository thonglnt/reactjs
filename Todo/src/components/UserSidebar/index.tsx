// Lib
import { Link } from 'react-router-dom';

// Components
import ListUser from '@components/ListUser';
import Sidebar from '@components/Sidebar';

// Types
import { IUser } from 'types/user';

// Constants
import { PATH } from '@constants/paths';

export type UserSidebarProps = {
  users: IUser[];
  onRemoveUser: (id: string, name: string) => void;
};

const UserSidebar = ({ users, onRemoveUser }: UserSidebarProps) => {
  return (
    <Sidebar>
      <h2 className='title sidebar-title'>List Users</h2>
      <ListUser onRemoveUser={onRemoveUser} listUser={users} />
      <Link to={PATH.HOME + PATH.REGISTER_USER}>
        <button className='button'>New user</button>
      </Link>
    </Sidebar>
  );
};

export default UserSidebar;
