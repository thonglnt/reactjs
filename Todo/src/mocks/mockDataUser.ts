// types
import { IUser } from 'types/user';

export const mockDataUser: IUser[] = [
  {
    id: '1',
    name: 'user1',
    avatar: 'https://haycafe.vn/wp-content/uploads/2022/03/Avatar-hai-1.jpg',
  },
  {
    id: '2',
    name: 'user2',
    avatar: 'https://tapchianhdep.com/wp-content/uploads/2021/05/hinh-avatar-hai-huoc-ngau.jpg',
  },
  {
    id: '3',
    name: 'user3',
    avatar: 'https://i.pinimg.com/736x/1c/44/41/1c444192c50fd9101640cdf99fd4a6c7.jpg',
  },
  {
    id: '4',
    name: 'user4',
    avatar:
      'https://1.bp.blogspot.com/-Ada3siBi-Z8/YS3asSghKxI/AAAAAAABVSc/EJKXWlcxH1shPt6rWc-q3gxvSk2EdyZyACLcBGAsYHQ/s0/Avatar-con-viet-vang-ngau-Taihinhanh-Vn%2B%25286%2529.jpg',
  },
];
