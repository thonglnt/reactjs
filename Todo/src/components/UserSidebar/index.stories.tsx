// Lib
import { Story } from '@storybook/react';
import { action } from '@storybook/addon-actions';

// Components
import UserSidebar, { UserSidebarProps } from '@components/UserSidebar';

// Mocks
import { mockDataUser } from '@mocks/mockDataUser';
import { BrowserRouter } from 'react-router-dom';

export default {
  title: 'UserSidebar',
  component: UserSidebar,
};

export const Sample: Story<UserSidebarProps> = () => (
  <BrowserRouter>
    <UserSidebar users={mockDataUser} onRemoveUser={action('onRemoveUser')} />
  </BrowserRouter>
);
