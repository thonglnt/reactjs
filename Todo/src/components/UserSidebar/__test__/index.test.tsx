// Lib
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import UserSidebar, { UserSidebarProps } from '@components/UserSidebar';

// Mocks
import { mockDataUser } from '@mocks/mockDataUser';

describe('Test UserSidebar component', () => {
  const initUserSidebar: UserSidebarProps = {
    users: mockDataUser,
    onRemoveUser: jest.fn(),
  };

  test('It should render correctly with initial Props', () => {
    const { container } = render(
      <BrowserRouter>
        <UserSidebar {...initUserSidebar} />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  test('It should match data for UserSidebar component', () => {
    render(
      <BrowserRouter>
        <UserSidebar {...initUserSidebar} />
      </BrowserRouter>,
    );

    expect(screen.getAllByRole('img')[0].getAttribute('src')).toEqual(
      'https://haycafe.vn/wp-content/uploads/2022/03/Avatar-hai-1.jpg',
    );
  });
});
