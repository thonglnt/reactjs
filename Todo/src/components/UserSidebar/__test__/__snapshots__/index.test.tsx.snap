// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserSidebar component It should render correctly with initial Props 1`] = `
<div>
  <aside
    class="sidebar-wrapper"
  >
    <h2
      class="title sidebar-title"
    >
      List Users
    </h2>
    <ul
      class="user-list"
    >
      <li
        class="user-item"
        data-testid="selected-id-user"
      >
        <img
          alt="Avatar"
          class="avatar"
          src="https://haycafe.vn/wp-content/uploads/2022/03/Avatar-hai-1.jpg"
        />
        user1
        <img
          alt="user name"
          class="icon icon-delete"
          data-testid="delete-user"
          src="IMAGE_MOCK"
        />
      </li>
      <li
        class="user-item"
        data-testid="selected-id-user"
      >
        <img
          alt="Avatar"
          class="avatar"
          src="https://tapchianhdep.com/wp-content/uploads/2021/05/hinh-avatar-hai-huoc-ngau.jpg"
        />
        user2
        <img
          alt="user name"
          class="icon icon-delete"
          data-testid="delete-user"
          src="IMAGE_MOCK"
        />
      </li>
      <li
        class="user-item"
        data-testid="selected-id-user"
      >
        <img
          alt="Avatar"
          class="avatar"
          src="https://i.pinimg.com/736x/1c/44/41/1c444192c50fd9101640cdf99fd4a6c7.jpg"
        />
        user3
        <img
          alt="user name"
          class="icon icon-delete"
          data-testid="delete-user"
          src="IMAGE_MOCK"
        />
      </li>
      <li
        class="user-item"
        data-testid="selected-id-user"
      >
        <img
          alt="Avatar"
          class="avatar"
          src="https://1.bp.blogspot.com/-Ada3siBi-Z8/YS3asSghKxI/AAAAAAABVSc/EJKXWlcxH1shPt6rWc-q3gxvSk2EdyZyACLcBGAsYHQ/s0/Avatar-con-viet-vang-ngau-Taihinhanh-Vn%2B%25286%2529.jpg"
        />
        user4
        <img
          alt="user name"
          class="icon icon-delete"
          data-testid="delete-user"
          src="IMAGE_MOCK"
        />
      </li>
    </ul>
    <a
      href="/register"
    >
      <button
        class="button"
      >
        New user
      </button>
    </a>
  </aside>
</div>
`;
